hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java

```diff
@@ -1250,4 +1253,14 @@ public String toString() {
                 + ", lastIdLoaded=" + lastIdLoaded
                 + '}';
     }
+
+    /**
+     * @since 5.01
+     */
+    public void resetAgeStats() {
+        minAge = -1;
```

Comment:
The default value for `minAge` is `Long.MAX_VALUE` and `maxAge` is `Long.MIN_VALUE`. Why don't we reset to these values?

Reply:
Yes, that's how I implemented it initially, and it "didn't work", meaning that the assertion of [QueueResetAgeStatisticsTest:78](https://github.com/hazelcast/hazelcast/pull/19625/files#diff-7f18819cc4217b1f239cc97069f1cf366df81682daf9947e7ec6295a11a4f47bR78) failed, because `minAge` was `Long.MAX_VALUE` instead of `0`. I wanted to avoid this, because it would be unreasonable behavior from a user perspective: if `minAge` is `0` initially, then `minAge` should be `0` after reset as well (and not `Long.MAX_VALUE`).

It turned out that the difference between thte two flows (reading minAge initially vs. reading minAge after reset) is that when we read the queue stats first, without ever accessing the queue before, there is no `QeueContainer` created yet, therefore `QueueService#createLocalQueueStats()` [short-circuit returns an empty `LocalMapStatsImpl` object](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueService.java#L344). It means, the `minAge` seen by the caller is the default value of [LocalQueueStats#minAge](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/LocalQueueStatsImpl.java#L62) (which is `0`) and not `QueueContainer#minAge` (which defaults to `Long.MAX_VALUE`). 

Once I discovered this problem, I decided to use a special value (-1) in the `QueueContainer` to mark that there is no `minAge` received yet. I know it adds some complexity to the control flow, but I couldn't find any simpler solutions to the problem.

Reply:
Why not make https://github.com/hazelcast/hazelcast/blob/06b3d99e1379e7389434230a9d94a7dfc05bf967/hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueService.java#L344 return the default value of `Long.MIN_VALUE` so that there's no duality?

Reply:
Currently in 5.0 it returns 0 for a queue that was never touched. Changing it to Long.MAX_VALUE would be a slightly breaking change IMO.

Reply:
I think it's acceptable and probably not a very well defined behavior in any case.